src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RazorComponentRenameEndpoint.cs

```diff
@@ -293,12 +293,28 @@ private async Task<IReadOnlyList<TagHelperDescriptor>> GetOriginTagHelpersAsync(
             }
 
             var owner = syntaxTree.Root.LocateOwner(change);
+            if (owner == null)
+            {
+                Debug.Fail("Owner should never be null.");
+                return null;
+            }
+
             var node = owner.Ancestors().FirstOrDefault(n => n.Kind == SyntaxKind.MarkupTagHelperStartTag);
             if (node == null || !(node is MarkupTagHelperStartTagSyntax tagHelperStartTag))
             {
                 return null;
             }
 
+            // Ensure the rename action was invoked on the component name
+            // instead of a component parameter. This serves as an issue 
+            // mitigation till `textDocument/prepareRename` is supported 
+            // and we can ensure renames aren't triggered in unsupported
+            // contexts. (https://github.com/dotnet/aspnetcore/issues/26407)
+            if (owner.SpanStart != tagHelperStartTag.SpanStart)
```

Comment:
Does `tagHelperStartTag` include the `<` in `<MyComponent>`? If so does that work correctly if the owner is the name part of the tag? Might be better to use the `tagHelperStartTag.Name` span.

Reply:
So if I understood correctly, you're asking what happens if rename is invoked:

`<Survey|Prompt Title="jkldsajfl" />`

If so, that works correctly

![image](https://user-images.githubusercontent.com/14852843/94515620-c8e5d000-01d8-11eb-9cf2-7bb1cbbd985b.png)
![image](https://user-images.githubusercontent.com/14852843/94515685-f894d800-01d8-11eb-8757-f8424c51768c.png)

So to answer your initial question, `owner.SpanStart == tagHelperStartTag.SpanStart` because both the `tagHelperStartTag` & `owner` start at `<SurveyPrompt`

Reply:
I interpreted his question as being about what happens in the case of invoking rename on `|<SurveyPrompt`.

Reply:
![RenameBeforeTagStart](https://user-images.githubusercontent.com/14852843/94592036-b18cfe00-023d-11eb-87b1-d423a05528e4.gif)

We get a rename prompt for `<`. This isn't at all what we want, but the best we can do for now, is have it no-op. Proper solution requires `textDocument/prepareRename`

Reply:
Feels more correct to be looking at the Name property on the `MarkupTagHelperStartTag`

Reply:
+1 to what @NTaylorMullen. That is what I was getting at. Because,

> So to answer your initial question, owner.SpanStart == tagHelperStartTag.SpanStart because both the tagHelperStartTag & owner start at <SurveyPrompt

even though the above seems to work, it doesn't seem right to compare while including the `<` given what we are actually renaming is only the tag name.